src/library/BaseValidator.ts

```diff
@@ -107,4 +120,29 @@ export class BaseValidator {
     public static onlyId(repository: BaseRepository): RequestHandler[] {
         return BaseValidator.validationList({ id: BaseValidator.validators.id(repository) });
     }
+
+    /**
+     * formatData
+     *
+     * Formata a data informada do tipo string para um tipo Date
+     *
+     *
+     * @param data Recebe um string com a data para ser formada
+     */
+    public static formatDate(data: string): Date | undefined {
+        const dateItems: any[] = data.split('/');
+        const formatItems: string[] = 'dd/mm/yyyy'.split('/');
+        const dayIndex: number = formatItems.indexOf('dd');
+        const monthIndex: number = formatItems.indexOf('mm');
+        if (monthIndex < 0 || monthIndex >= 13) return undefined;
```

Comment:
Todas essas validações que retornar undefined da pra resolver com uma regex de data.

Reply:
foi o que eu te falei, o regex deixa passar algumas datas de dias como (35), ele contabiliza os dias a partir da data do mês., por exemplo: mês de dezembro tem 31 dias, se alguém setar a data 35 por engano, ele vai salvar no banco 04/janeiro.

Mesma coisa para o mês, ele deixa salvar o mês 00

Reply:
Regex simples que resolve a maioria dos casos:
[1-31]\/[1-12]\/\d{4}

Regex complexa:
https://stackoverflow.com/questions/15491894/regex-to-validate-date-formats-dd-mm-yyyy-dd-mm-yyyy-dd-mm-yyyy-dd-mmm-yyyy